pkg/sdk/schemas_integration_test.go

```diff
@@ -0,0 +1,370 @@
+		})
+		require.NoError(t, err)
+
+		clonedSchemaID := NewSchemaIdentifier(db.Name, randomAccountObjectIdentifier(t).name)
+		err = client.Schemas.Create(ctx, clonedSchemaID, &CreateSchemaOptions{
+			Comment: String(comment),
+			Clone: &Clone{
+				SourceObject: schemaID,
+			},
+		})
+		require.NoError(t, err)
+
+		s, err := client.Schemas.ShowByID(ctx, schemaID)
+		require.NoError(t, err)
+
+		cs, err := client.Schemas.ShowByID(ctx, clonedSchemaID)
+		require.NoError(t, err)
+		assert.Equal(t, *s.Comment, *cs.Comment)
+
+		t.Cleanup(func() {
+			err = client.Schemas.Drop(ctx, schemaID, nil)
+			require.NoError(t, err)
+			err = client.Schemas.Drop(ctx, clonedSchemaID, nil)
+			require.NoError(t, err)
+		})
+	})
+
+	t.Run("with tags", func(t *testing.T) {
+		tagName := randomString(t)
+		tagID := NewAccountObjectIdentifier(tagName)
+		_, err := client.exec(ctx, fmt.Sprintf(`CREATE TAG "%s"`, tagName))
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			_, err := client.exec(ctx, fmt.Sprintf(`DROP TAG "%s"`, tagName))
+			require.NoError(t, err)
+		})
+
+		schemaID := NewSchemaIdentifier(db.Name, randomAccountObjectIdentifier(t).name)
+		tagValue := randomString(t)
+		err = client.Schemas.Create(ctx, schemaID, &CreateSchemaOptions{
+			Tag: []TagAssociation{
+				{
+					Name:  tagID,
+					Value: tagValue,
+				},
+			},
+		})
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			err = client.Schemas.Drop(ctx, schemaID, nil)
+			require.NoError(t, err)
+		})
+
+		tv, err := client.SystemFunctions.GetTag(ctx, tagID, schemaID, ObjectTypeSchema)
+		require.NoError(t, err)
+		assert.Equal(t, tagValue, tv)
+	})
+}
+
+func TestInt_SchemasAlter(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	db, cleanupDb := createDatabase(t, client)
+	t.Cleanup(cleanupDb)
+
+	t.Run("rename to", func(t *testing.T) {
+		schema, _ := createSchema(t, client, db)
+		newID := NewSchemaIdentifier(db.Name, randomString(t))
+		err := client.Schemas.Alter(ctx, schema.ID(), &AlterSchemaOptions{
+			NewName: newID,
+		})
+		require.NoError(t, err)
+		s, err := client.Schemas.ShowByID(ctx, newID)
+		require.NoError(t, err)
+		assert.Equal(t, newID, s.ID())
+		t.Cleanup(func() {
+			err = client.Schemas.Drop(ctx, newID, nil)
+			require.NoError(t, err)
+		})
+	})
+
+	t.Run("swap with", func(t *testing.T) {
+		schema, cleanupSchema := createSchema(t, client, db)
+		t.Cleanup(cleanupSchema)
+
+		swapSchema, cleanupSwapSchema := createSchema(t, client, db)
+		t.Cleanup(cleanupSwapSchema)
+
+		table, _ := createTable(t, client, db, schema)
+		err := client.Schemas.Alter(ctx, schema.ID(), &AlterSchemaOptions{
+			SwapWith: swapSchema.ID(),
+		})
+		require.NoError(t, err)
+
+		schemaDetails, err := client.Schemas.Describe(ctx, swapSchema.ID())
+		require.NoError(t, err)
+		assert.Equal(t, 1, len(schemaDetails))
+		assert.Equal(t, "TABLE", schemaDetails[0].Kind)
+		assert.Equal(t, table.Name, schemaDetails[0].Name)
+
+		t.Cleanup(func() {
```

Comment:
1. I think we should discuss cleanup in such case (because many other tests are not doing it in such one, but TBH I don't think that's bad).
2. This cleanup is too late, we may have error earlier which will prevent this cleanup to be registered (`require.NoError(t, err)` after describe)

Reply:
Right, I'll move it up